src/shared/Atlassian.Bitbucket/BitbucketHostProvider.cs

```diff
@@ -101,42 +150,28 @@ public async Task<ICredential> GetCredentialAsync(InputArguments input)
 
                 if (SupportsBasicAuth(authModes))
                 {
-                    _context.Trace.WriteLine("Checking for credentials...");
-                    ICredential credential = _context.CredentialStore.Get(credentialService, input.UserName);
-                    if (credential is null)
-                    {
-                        // We don't have any credentials to use at all! Start with the assumption of no 2FA requirement
-                        // and capture username and password via an interactive prompt.
-                        credential = await _bitbucketAuth.GetBasicCredentialsAsync(targetUri, input.UserName);
-                        if (credential is null)
-                        {
-                            throw new Exception("User cancelled authentication prompt.");
-                        }
-                    }
+                    _context.Trace.WriteLine("Prompt for Basic Auth...");
 
-                    // Either we have an existing credential (user/pass OR some form of token [PAT or AT]),
-                    // or we have a freshly captured user/pass. Regardless, we must check if these credentials
-                    // pass and two-factor requirement on the account.
-                    _context.Trace.WriteLine("Checking if two-factor requirements for stored credentials...");
+                    var basicCredentials = await GetBasicCredentialsInteractive(input, authModes);
 
-                    bool requires2Fa = await RequiresTwoFactorAuthenticationAsync(credential, authModes);
-                    if (!requires2Fa)
+                    if(await ValidateCredentialsWork(input, basicCredentials, authModes))
```

Comment:
The prevailing style in GCM is a space between the `if` keyword and the `()`, i.e, `if (true)` and not `if(true)`.
Please update instances in this PR